# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/credentialmanagement/identity/WebIdentityChild.h
# Commit: 0c5428075ca4
# Full Hash: 0c5428075ca4aa6b5270350247d43217fa5d0feb
# Author: Benjamin VanderSloot <bvandersloot@mozilla.com>
# Date: 2025-06-07 08:31:23
# Regressor Bug: 1969017
# File Overlap Count: 2
# Description:
#   Bug 1969017 - Refactor FedCM to use a managed actor rather than tagging onto WindowGlobalParent - r=anti-tracking-reviewers,emz
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D251617
# ==============================================================================

diff -r 9e7fe3dbea01 -r 0c5428075ca4 dom/credentialmanagement/identity/WebIdentityChild.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/credentialmanagement/identity/WebIdentityChild.h	Fri Jun 06 14:38:54 2025 +0000
@@ -0,0 +1,35 @@
+/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim: set ts=8 sts=2 et sw=2 tw=80: */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef mozilla_dom_WebIdentityChild_h
+#define mozilla_dom_WebIdentityChild_h
+
+#include "mozilla/dom/PWebIdentity.h"
+#include "mozilla/dom/PWebIdentityChild.h"
+
+namespace mozilla::dom {
+
+class WebIdentityHandler;
+
+class WebIdentityChild final : public PWebIdentityChild {
+ public:
+  NS_INLINE_DECL_REFCOUNTING(WebIdentityChild, override);
+
+  WebIdentityChild() = default;
+
+  void ActorDestroy(ActorDestroyReason why) override;
+
+  void SetHandler(WebIdentityHandler* aHandler);
+
+ private:
+  ~WebIdentityChild() = default;
+
+  WebIdentityHandler* mHandler{};
+};
+
+}  // namespace mozilla::dom
+
+#endif  // mozilla_dom_WebIdentityChild_h